src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java

```diff
@@ -47,44 +48,33 @@ public static FeedbackQuestionsLogic inst() {
     }
     
     public void createFeedbackQuestion(FeedbackQuestionAttributes fqa)
-            throws InvalidParametersException {
+            throws InvalidParametersException, EntityDoesNotExistException, EntityAlreadyExistsException {
         
-        String feedbackSessionName = fqa.feedbackSessionName;
-        String courseId = fqa.courseId;
-        List<FeedbackQuestionAttributes> questions = null;
-        
-        try {
-            questions = getFeedbackQuestionsForSession(feedbackSessionName, courseId);
-        } catch (EntityDoesNotExistException e) {
-            Assumption.fail("Session disappeared.");
-        }
-        if (fqa.questionNumber < 0) {
-            fqa.questionNumber = questions.size() + 1;
-        }
-        adjustQuestionNumbers(questions.size() + 1, fqa.questionNumber, questions);
-        createFeedbackQuestionNoIntegrityCheck(fqa, fqa.questionNumber);
+        fqa.removeIrrelevantVisibilityOptions();
+        adjustQuestionNumbersAndCreateQuestion(fqa, -1);
     }
     
     /**
      * Used for creating initial questions only.
-     * Does not check if feedback session exists.
-     * Does not check if question number supplied is valid(does not check for clashes, or make adjustments)
+     * Does not check if question number supplied is valid (does not check for clashes, or make adjustments)
      * @param fqa
      * @param questionNumber
      * @throws InvalidParametersException
+     * @throws EntityDoesNotExistException if the session does not exist
      */
     public FeedbackQuestionAttributes createFeedbackQuestionNoIntegrityCheck(
-            FeedbackQuestionAttributes fqa, int questionNumber) throws InvalidParametersException {
+            FeedbackQuestionAttributes fqa, int questionNumber) throws InvalidParametersException,
```

Comment:
Could write `throws InvalidParametersException` on the next line (with `EntityDoesNotExistException`)